IoTPiCar.py

```diff
@@ -41,163 +55,135 @@ def calculate_cycles_from_distance(distance: int):
+    
+    if(start_angle >= SERVO_MAX_ANGLE):
+        servo_delta = -SERVO_STEP
+    elif(start_angle <= SERVO_MIN_ANGLE):
+        servo_delta = SERVO_STEP
+    else:
+        servo_delta = 0
+
+    while ((start_angle == SERVO_MIN_ANGLE and next_angle <= SERVO_MAX_ANGLE) or
+          (start_angle == SERVO_MAX_ANGLE and next_angle >= SERVO_MIN_ANGLE)):
+        
+        distance = get_status_at(next_angle, servo_speed=servo_speed)
+        #print('current angle', next_angle)
         isDetected = 1 if distance <= detection_distance and not distance == -2 else 0
-        detection_list.append(isDetected)
-        current_angle += servo_delta
-
-        if(current_angle > SERVO_MAX_ANGLE):
-            servo_delta = -servo_delta
-            print('servo max angle')
-        if(current_angle < SERVO_MIN_ANGLE):
-            print('servo min angle', servo_delta)
-            servo_delta = abs(servo_delta)
-
-        print('current angle', current_angle)
-
-    print('detection list', detection_list)
-    servo.set_angle(SERVO_ZERO_ANGLE)
-
-    return True
-# def self_drive():
-#     servo.set_angle(0)
-#     picar.forward(constants.WHEEL_POWER)
-#     isObjectFound = False
-#     while not isObjectFound:
-#         print('running wheels! lol')
-#         scan_info = scan_step()
-#         if(scan_info):
-#             scan_subset = utils.detectionSubset(scan_info)
-
-#         if(scan_info):
-#             print('scaninfo %s:', scan_info)
-#             print('subset %s', scan_subset)
-#             wasItemDetected = utils.isDetected(scan_subset)
-#             if(wasItemDetected and wasItemDetected):
-#                 print('is detected halting')
-#                 isObjectFound = True
-#                 picar.stop()
-#     picar.stop()
-#     servo.set_angle(0)
-
-def decide_movement(scan_results: list):
-    detection = scan_results[1]
-    forward_domain = detection[3:7]
-    left_domain = detection[0:2]
-    right_domain = detection[8:10]
-
-    print(scan_results[1])
-    isSomethingAhead = sum(forward_domain) >= 1
-    isSomethingLeft = sum(left_domain) >= 1
-    isSomethingRight = sum(right_domain) >= 1
-    if(isSomethingAhead and isSomethingLeft):
-        move(5, movement_type='rotation', direction='right')
-    if(isSomethingAhead and isSomethingRight):
-        move(5, movement_type='rotation', direction='left')
-
-    detection_distances = []
-    safe_forward_cycles = 0
-    print('scan result', scan_results[0][3:8])
-    for scan_result in scan_results[0][3:7]:
-        if(not scan_result['distance'] == -2):
-            detection_distances.append(scan_result['distance'])
+        scan_info.append({'distance': distance, 'angle': next_angle, 'detection': isDetected})
+        detection_list.append(isDetected)       
+        next_angle = get_servo_angle() + servo_delta
+ 
+    #print('One Sweep detection list:', detection_list) 
+    return (scan_info, detection_list)
+
+def decide_movement(sweep_info, Isdetected):
```

Comment:
the is[item] pattern is more for single booleans, than lists of booleans. detections would work better.

Reply:
sweep_info already has this info but I am not an expert in python on how to extract it out of that data structure without for loop. Any help is appreciated. If we could do that then this Isdetected can be completely removed

Reply:
I would also like to avoid the second return 'detection_list' of the function perform_one_sweep if you have a better way